server/src/test/java/com/linecorp/centraldogma/server/internal/httpapi/CommitServiceV1Test.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.server.internal.httpapi;
+
+import static net.javacrumbs.jsonunit.fluent.JsonFluentAssert.assertThatJson;
+
+import java.net.InetSocketAddress;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.client.HttpClientBuilder;
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.centraldogma.testing.CentralDogmaRule;
+
+public class CommitServiceV1Test {
+
+    // TODO(minwoox) replace this unit test using nested structure in junit 5
+    // Rule is used instead of ClassRule because the listCommits is affected by other unit tests.
+    @Rule
+    public final CentralDogmaRule dogma = new CentralDogmaRule();
```

Comment:
Alternatively, you can use a `static` field here and use a different repository for each test method, using the `TestName` rule.